pkg/agent/cniserver/server.go

```diff
@@ -165,14 +165,14 @@ func updateResultIfaceConfig(result *current.Result, defaultGateways []net.IP) {
 	} else {
 		result.Routes = []*cnitypes.Route{}
 	}
-	for _, gw := range defaultGateways {
-		if (!foundV4DefaultRoute) && (gw.To4() != nil) {
-			_, defaultV4RouteDstNet, _ := net.ParseCIDR(defaultV4RouteDst)
-			result.Routes = append(result.Routes, &cnitypes.Route{Dst: *defaultV4RouteDstNet, GW: gw})
-		} else if (!foundV6DefaultRoute) && (gw.To4() == nil) {
-			_, defaultV6RouteDstNet, _ := net.ParseCIDR(defaultV6RouteDst)
-			result.Routes = append(result.Routes, &cnitypes.Route{Dst: *defaultV6RouteDstNet, GW: gw})
-		}
+
+	if (!foundV4DefaultRoute) && (defaultIPv4Gateway != nil) && (defaultIPv4Gateway.To4() != nil) {
```

Comment:
`defaultIPv4Gateway` should be an IPv4 address, so the check for IPv4 address is not needed if it is not nil.